riscv/src/runtime.rs

```diff
@@ -148,6 +148,29 @@ impl Runtime {
         self.syscalls.contains_key(&s)
     }
 
+    pub fn with_keccak(mut self) -> Self {
+        self.add_submachine(
+            "std::machines::hash::keccakf::KeccakF",
+            None,
+            "keccakf",
+            [format!(
+                "instr keccakf ~ keccakf.keccakf {};", // operation
+                instr_register_params(0, 2, 2)         // 2 pointers as both input and output
+            )],
+            0,
+            std::iter::once("keccakf;".to_string()) // must be called at least once
+                .chain((0..2).map(|i| format!("{} <=X= 0;", reg(i)))), // zero out output registers
+        );
+
+        // The keccakf syscall has a two arguments passed on x10 and x11,
+        // the memory address of the 25 field element input array
```

Comment:
50 field elements

Reply:
Alrighty, I'll make all input and output u32 then.

Reply:
what is this related to?

Reply:
Ignore my comment for this. It should still be u64, but 25 field elements for the input and 25 field elements for the output, so a total of 50 field elements. I modified the comments to make this clear.